### PR TITLE
fix: add sandbox username to command JSON

### PR DESCRIPTION
**Pull request by shetzel**
### What does this PR do?
Adds a `SandboxUsername` property to the JSON response for `org sandbox create`, `org sandbox refresh` and `org sandbox resume`.

### What issues does this PR fix or reference?
@W-15818633@

### CODE REVIEW

File: src/shared/sandboxCommandBase.ts [hunk 1]
```diff
@@ -230,6 +237,32 @@ export abstract class SandboxCommandBase<T> extends SfCommand<T> {
     }
   }
 
+  // Gets the SandboxName either from the request data or the SandboxProcessObject
+  protected getSandboxName(): string | undefined {
+    return this.sandboxRequestData?.sandboxProcessObject.SandboxName ?? this.latestSandboxProgressObj?.SandboxName;
+  }
+
+  // Not sure why the lint rule is complaining since it's definitely called.
+  // eslint-disable-next-line class-methods-use-this
```
Comment (WillieRuemmele) [inline src/shared/sandboxCommandBase.ts:246]: it's because the method doesn't use `this` in it

File: src/shared/sandboxCommandBase.ts [hunk 2]
```diff
@@ -230,6 +237,32 @@ export abstract class SandboxCommandBase<T> extends SfCommand<T> {
...
+    if (this.sandboxUsername) {
+      sbxUsername = this.sandboxUsername;
+    } else {
+      const prodOrgUsername = this.prodOrg?.getUsername();
+      const sandboxName = this.getSandboxName();
+      if (prodOrgUsername && sandboxName) {
+        sbxUsername = this.getSandboxUsername(prodOrgUsername, sandboxName);
+      }
+    }
+    return Object.assign({}, this.latestSandboxProgressObj, { SandboxUsername: sbxUsername });
```
Comment (WillieRuemmele) [inline src/shared/sandboxCommandBase.ts:263]: NIT: `return {...this.latestSandboxProgressObj, SandboxUsername: sbxUsername}`

File: test/nut/sandboxCreate.nut.ts [hunk 3]
```diff
@@ -85,14 +85,15 @@ describe('Sandbox Create', () => {
     const sbxName = 'createSbx1';
     const sbxLicenseType = 'Developer';
     const sbxProcess = getSandboxProcess({ SandboxName: sbxName });
+    const expectedCmdResponse = Object.assign({}, sbxProcess, { SandboxUsername: `${hubOrgUsername}.${sbxName}` });
```
Comment (WillieRuemmele) [inline test/nut/sandboxCreate.nut.ts:88]: NIT: `object.assign` again, but you could also call `getSandboxUsername()` too 🤷‍♂️

**Review by WillieRuemmele — APPROVED** [review]

**Comment by WillieRuemmele** [comment]
sandbox NUTs run https://github.com/salesforcecli/plugin-org/actions/runs/9214331936

File: src/shared/sandboxCommandBase.ts (hunk 1 above)
Comment (shetzel) [reply src/shared/sandboxCommandBase.ts:246]: Ah, that's right, thank you!

**Comment by WillieRuemmele** [comment]
### QA Notes
---
❌ -> ✅ : the sandbox username should not be present before the sandbox has been authorized.
... once the sandbox has authorized
```bash
 ➜  ../../oss/plugin-org/bin/run.js org resume sandbox --use-most-recent -o na40 --json

{
  "status": 0,
  "result": {
    "attributes": {
      "type": "SandboxProcess",
      "url": "/services/data/v61.0/tooling/sobjects/SandboxProcess/xxx"
    },
    "Id": "xxx",
    "Status": "Completed",
    "SandboxName": "WillieSBX3",
    "SandboxInfoId": "xxx",
    "LicenseType": "DEVELOPER",
    "CreatedDate": "2024-05-30T19:27:57.000+0000",
    "CopyProgress": 100,
    "SandboxOrganization": "00",
    "SourceId": null,
    "Description": null,
    "EndDate": "2024-05-30T19:42:01.000+0000",
    "SandboxUsername": "xxx"
```
✅ : cleared out of cache
🟨 : waiting on created sandboxes 24 hours to be able to refresh (3 pm MST)

**Pull request merged**